analysisCode/src/include/TruthEvent.h

```diff
@@ -1,19 +1,188 @@
 #ifndef TRUTHEVENT_H
 #define TRUTHEVENT_H
 
+#include <eicsmear/erhic/EventPythia.h>
+#include <eicsmear/smear/EventS.h>
 
+#include <fastjet/ClusterSequence.hh>
+#include <fastjet/Selector.hh>
+#include <fastjet/contrib/SoftDrop.hh>
+#include <vector>
+
+#include "JetDef.h"
+#include "SoftDropJetDef.h"
+
+using namespace std;
+using namespace fastjet;
 
 class TruthEvent {
   
  public:
+
   TruthEvent();
+  //  TruthEvent(erhic::EventPythia* ev);
   ~TruthEvent(){}
 
- private:
+  // member functions //
+  void Set_truthEvent(erhic::EventPythia* ev);
+  void Set_truthPart(const Particle *part);
+  void Set_truthJet(PseudoJet jet);
+  void Set_groomed_truthJet(PseudoJet jet);
+
+  int Get_processId() const {return processId;}
+  int Get_eventNumber() const {return eventNumber;}
+  double Get_true_q2() const {return true_q2;}
+  double Get_true_x() const {return true_x;}
+  double Get_q2() const {return q2;}
+
+  
+  double Get_part_px()  const {return part_px;}
+  double Get_part_py()  const {return part_py;}
```

Comment:
I don't think we should have particle information in the `TruthEvent` class. In principle, I don't think we will ever need particle information other than for the jet reconstruction, since the resulting jets contain the information about their constituents.

Reply:
This makes sense, I will remove the corresponding member variables and functions.